include/dt-bindings/dma/stm32_dma.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_DMA_STM32_DMA_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_DMA_STM32_DMA_H_
+
+#define STM32_DMA_MEMORY_TO_MEMORY        0
+#define STM32_DMA_MEMORY_TO_PERIPH        1
+#define STM32_DMA_PERIPH_TO_MEMORY        2
+
+#define STM32_DMA_PRIORITY_VERYHIGH       0
+#define STM32_DMA_PRIORITY_HIGH           1
+#define STM32_DMA_PRIORITY_MEDIUM         2
+#define STM32_DMA_PRIORITY_LOW            3
+
+#define STM32_DMA_SRC_NO_INCREMENT        0
+#define STM32_DMA_SRC_INCREMENT           1
+
+#define STM32_DMA_DST_NO_INCREMENT        0
+#define STM32_DMA_DST_INCREMENT           1
+
+#define STM32_DMA_NORMAL_MODE             0
+#define STM32_DMA_CIRCULAR_MODE           1
```

Comment:
Actually, there is one more mode: Double Buffer mode, based on STM32F4's RM. Will you consider to add that one?

Reply:
I would like to add it but this must be amended after the dma api supports it. And API is hard to change due to the stability requirement of zephyr. Maybe zephyr's dma api needs an application specified pointer pointing to a driver specified structure to convey some special configurations like if the dual buffer mode is used. :worried: